scripts/update_translations.js

```diff
@@ -0,0 +1,258 @@
+    // Create the file if it does not exist
+    if (!fs.existsSync(poFile)) {
+      fs.writeFileSync(poFile, '');
+    }
+
+    // Merge the `.pot` file with the existing translations
+    execSync(`msgmerge ${poFile} ${potFile} -o ${poFile}`, { stdio: 'inherit' });
+  });
+}
+
+/**
+ * Checks if all `.po` files have all messages from the `.pot` file.
+ * Allows untranslated messages if `strict` is false.
+ * @param {boolean} strict - If true, does not allow untranslated messages.
+ * @returns {boolean} True if an error happened or if there are any differences
+ */
+function checkPoTranslations(strict = false) {
+  console.log(`⏳ Checking translations...`);
+
+  const localeDir = 'locale';
+  const potFile = path.join(localeDir, 'texts.pot');
+  const subfolders = fs.readdirSync(localeDir)
+    .filter((subfolder) => fs.statSync(path.join(localeDir, subfolder)).isDirectory());
+  let errorHappened = false;
+
+  subfolders.forEach((subfolder) => {
+    const poFile = path.join(localeDir, subfolder, 'texts.po');
+    const cmd = strict ? `msgcmp ${poFile} ${potFile}` : `msgcmp --use-untranslated ${poFile} ${potFile}`;
+    try {
+      // If there are any differences, the command will throw an error
+      execSync(cmd, { stdio: 'inherit' });
+    } catch (error) {
+      console.error(`Error checking ${poFile}:`, error);
+      errorHappened = true;
+    }
+  });
+
+  if (errorHappened) {
+    console.warn('❌ Some translations are missing. Please review the changes.');
+  }
+  return errorHappened
+}
+
+/**
+ * Checks for any fuzzy tags in the `.po` files, as they indicate human review is necessary
+ * @see https://www.gnu.org/software/gettext/manual/html_node/Fuzzy-Entries.html
+ * @returns {boolean} True if any fuzzy tags were found
+ */
+function checkFuzzyTags() {
+  console.log(`⏳ Checking for fuzzy tags...`);
+
+  const localeDir = 'locale';
+  const subfolders = fs.readdirSync(localeDir)
+    .filter((subfolder) => fs.statSync(path.join(localeDir, subfolder))
+      .isDirectory());
+  let containsFuzzy = false;
+
+  subfolders.forEach((subfolder) => {
+    const poFile = path.join(localeDir, subfolder, 'texts.po');
+    const content = fs.readFileSync(poFile, 'utf-8');
+    if (content.includes('fuzzy')) {
+      containsFuzzy = true;
+    }
+  });
+
+  if (containsFuzzy) {
+    console.warn(`❌: Fuzzy translations found in '.po' files. Please review the changes.`);
+  }
+
+  return containsFuzzy;
+}
+
+/**
+ * Checks for filesystem changes using `git status`. If any changes are found, it means the
+ * translation files have changed, and this may trigger a CI fail when running inside the CI.
+ * @returns {boolean} True if changes were found, or if an error happened
+ */
+function checkForChanges() {
+  console.log(`⏳ Checking for filesystem changes after i18n execution...`);
+  let statusResult = '';
+  let hasChanges = false;
+  try {
+    statusResult = execSync('git status --porcelain', { encoding: 'utf-8' });
```

Comment:
Is it checking changes only in translation files?

Reply:
This checks for changes in the whole repository. However, the results were not being consistent across operational systems.

So I made this validation not fail the validation on https://github.com/HathorNetwork/hathor-wallet-mobile/pull/666/commits/78fbdb2331ed86b21032d4119d3a1b2be8a6efe0 . More information on this relaxation of rules is available on the updated PR description.